src/openssl/mod.rs

```diff
@@ -765,6 +1130,69 @@ mod test {
         assert_eq!(hash_bytes, expected_hash);
     }
 
+    /// Test vector for `SECP256K1-SHA256-SVDW` cipher suite
+    /// ASCII: "sample"
+    /// Oracle source: Generated by direct sage implementation https://github.com/mimblewimble/secp256k1-zkp/blob/master/sage/shallue_van_de_woestijne.sage
+    #[test]
+    fn test_hash_to_point_svdw_1() {
+        let mut vrf = ECVRF::from_suite(CipherSuite::SECP256K1_SHA256_SVDW).unwrap();
+
+        // Public key
+        let x = hex::decode("c9afa9d845ba75166b5c215767b1d6934e50c3db36e89b127b8a622b120f6721")
+            .unwrap();
+        let secret_key = BigNum::from_slice(&x).unwrap();
+        let public_key_point = vrf.derive_public_key_point(&secret_key).unwrap();
+
+        // Data to be hashed with svdw (ASCII "sample")
+        let data = hex::decode("73616d706c65").unwrap();
+        let z = BigNum::from_u32(1).unwrap();
+        let hashed_point = vrf
+            .hash_to_point_svdw(&public_key_point, &data, &z)
+            .unwrap();
+        let expected_vector = BigNum::from_hex_str(
+            "04bf6b0a69e608da4d9f9f1be84f5f7fd92e097f4467993a77758c4d603be1ba73a0175afbae1083465d6adf240802e8a65b63e330de60941ae513ffffd3d21740"
+        ).unwrap().to_vec();
+        assert_eq!(expected_vector.len(), 65);
+        let expected_point =
+            EcPoint::from_bytes(&vrf.group, &expected_vector, &mut vrf.bn_ctx).unwrap();
+        assert!(hashed_point
+            .eq(&vrf.group, &expected_point, &mut vrf.bn_ctx)
+            .unwrap());
+    }
+
+    /// Test vector for `SECP256K1-SHA256-SVDW` cipher suite
+    /// ASCII: "some long message to verify correctness of svdw implementation"
+    /// Oracle source: Generated by direct sage implementation https://github.com/mimblewimble/secp256k1-zkp/blob/master/sage/shallue_van_de_woestijne.sage
+    /// But the oracle source does not follow the standardized sign0 method discussed in https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-04.
```

Comment:
However, the oracle source does not follow the standardized sign0 method discussed in https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-04.

Reply:
Resolved